Source/SwiftLintBuiltInRules/Rules/Lint/BalancedXCTestLifecycleRule.swift

```diff
@@ -8,6 +8,39 @@ struct BalancedXCTestLifecycleRule: Rule {
         identifier: "balanced_xctest_lifecycle",
         name: "Balanced XCTest Life Cycle",
         description: "Test classes must implement balanced setUp and tearDown methods",
+        rationale: """
+        The `setUp` method of `XCTestCase` can be used to set up variables and resources before \
+        each test is run (or with the `class` variant, before all tests are run).
+
+        The memory model for XCTestCase objects is non-obvious. An instance of the `XCTestCase` \
```

Comment:
The memory model is important as well. However, this rule is about having a `tearDown` for every `setUp` method implemented for the sake of cleaning up/resetting resources that could otherwise cause effects on other tests (e.g. changed environment variables). Enforcing a `tearDown` at least makes you think about steps to perform after a test has run in order to leave everything in a consistent state and treat each test method as isolated. Basically, what's mentioned in the last sentence.